### DISCUSSION TITLE
"Inline" ky extend?

### DISCUSSION THREAD

**Discussion by filipjakov** [sindresorhus/ky#639] [Q&A] [answered]
I have the following situation:

- monorepo with multiple apps
- decided to extract the data layer into a separate monorepo package (lets call it lib)
- would define the "baseline" ky instance in that lib

since there could be env/app specific overrides for the ky client in each app (prefixUrl, hooks, ...), I'd like to have an "inline" extend -> the idea would be to import the ky client into the app of usage, do specific overrides (extends). It would apply automatically from the perspective of the lib.

TLDR: As far as I figured, `ky.extend` creates a new instance. I could use a "singleton" instance in this situation that could be extendable from for each app that uses the lib.

is this possible? is there a better approach to this problem?

**Comment by sholladay** [comment, marked as answer]
It might help if you could show an example of how you want it to work.

From what I can tell, the current `.extend()` mechanism would work ok for your situation. You can make an instance with custom options for each app. If they need to share options, you could make a root instance that they all extend from.

**Comment by filipjakov** [comment]
Thanks for getting back.

Your answer makes sense, but I forgot some crucial info 😅 I'm using a hook generator library (kubb for React Query) in that monorepo package, and it needs a client instance (ky) to point to when it generates the hooks queries.

Will try to create a simple example to share
